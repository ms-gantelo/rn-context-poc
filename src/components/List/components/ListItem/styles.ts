import { css } from '@emotion/native';

export const styles = {
  container: css`
    padding: 10px 15px;
    justify-content: space-between;
    flex-direction: row;
    align-items: center;
    background: white;
    margin: 3px 0;
    box-shadow: 0 3px 3px rgba(0, 0, 0, 0.05);
  `,
  dataContainer: css`
    flex-direction: row;
    align-items: center;
  `,
  link: css`
    color: blue;
  `,
  image: css`
    margin-right: 10px;
    height: 25px;
    width: 25px;
  `,
  artistName: css`
    font-size: 13px;
    font-weight: bold;
  `,
  count: css`
    font-size: 10px;
  `
};
