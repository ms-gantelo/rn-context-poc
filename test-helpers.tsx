import { IArtist } from './src/store/artist/state';
import { Deps } from './src/store/types';

export const getMockDeps = (): Deps =>
  ({
    apiService: { request: jest.fn().mockResolvedValue({ json: () => ({}) }) },
    stateSnapshot: { set: jest.fn(), get: jest.fn().mockReturnValue({}) }
  } as any);

export const getArtist_1 = (): IArtist => ({
  idArtist: '1',
  intBornYear: 1970,
  strArtist: 'Freddie Mercuty',
  strArtistThumb: 'thumb.png',
  strArtistBanner: 'banner.png',
  strArtistFanart: 'fanart.png',
  strBiographyEN: 'this is a bio',
  strCountry: 'USA',
  strStyle: 'rock'
});
