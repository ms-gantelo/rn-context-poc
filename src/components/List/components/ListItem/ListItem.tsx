import React, { memo } from 'react';
import { View, Text, Image } from 'react-native';

import { styles } from './styles';
import { TouchableOpacity } from 'react-native';
import { IArtist } from '../../../../store/artist/state';

export interface Props {
  artist: IArtist;
  albumCount: number;
  onPress: () => void;
}

const ListItem = ({ artist, albumCount, onPress }: Props) => {
  return (
    <TouchableOpacity onPress={onPress} testID="list-item">
      <View style={styles.container}>
        <View style={styles.dataContainer}>
          <Image
            style={styles.image as any}
            source={{
              uri:
                artist.strArtistThumb ||
                'https://www.dovercourt.org/wp-content/uploads/2019/11/610-6104451_image-placeholder-png-user-profile-placeholder-image-png.jpg'
            }}
          />
          <View>
            <Text style={styles.artistName}>{artist.strArtist}</Text>
            <Text style={styles.count}>{albumCount ? `${albumCount} albums` : ''}</Text>
          </View>
        </View>
        <Text style={styles.link}>view details</Text>
      </View>
    </TouchableOpacity>
  );
};

export default memo(ListItem);
