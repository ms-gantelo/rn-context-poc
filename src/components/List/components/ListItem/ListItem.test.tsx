import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';

import ListItem, { Props } from './ListItem';
import { getArtist_1 } from '../../../../../test-helpers';

describe('ListItem', () => {
  let props: Props;
  beforeEach(() => {
    props = {
      artist: getArtist_1(),
      albumCount: 0,
      onPress: jest.fn()
    };
  });

  it('should render without an album count', () => {
    const { toJSON } = render(<ListItem {...props} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render with an album count', () => {
    const { toJSON } = render(<ListItem {...props} albumCount={5} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render without a thumb', () => {
    const { toJSON } = render(<ListItem {...props} artist={{ ...getArtist_1(), strArtistThumb: null }} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should callback on press', async () => {
    const { findByTestId } = render(<ListItem {...props} artist={{ ...getArtist_1(), strArtistThumb: null }} />);
    const item = await findByTestId('list-item');
    fireEvent.press(item);
    expect(props.onPress).toBeCalled();
  });
});
