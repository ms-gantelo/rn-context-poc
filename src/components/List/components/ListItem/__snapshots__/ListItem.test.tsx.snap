// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render with an album count 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="list-item"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 3,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 3,
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
        "shadowColor": "rgba(0, 0, 0, 0.05)",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 3,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "thumb.png",
          }
        }
        style={
          Object {
            "height": 25,
            "marginRight": 10,
            "width": 25,
          }
        }
      />
      <View>
        <Text
          style={
            Object {
              "fontSize": 13,
              "fontWeight": "bold",
            }
          }
        >
          Freddie Mercuty
        </Text>
        <Text
          style={
            Object {
              "fontSize": 10,
            }
          }
        >
          5 albums
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "blue",
        }
      }
    >
      view details
    </Text>
  </View>
</View>
`;

exports[`ListItem should render without a thumb 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="list-item"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 3,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 3,
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
        "shadowColor": "rgba(0, 0, 0, 0.05)",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 3,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://www.dovercourt.org/wp-content/uploads/2019/11/610-6104451_image-placeholder-png-user-profile-placeholder-image-png.jpg",
          }
        }
        style={
          Object {
            "height": 25,
            "marginRight": 10,
            "width": 25,
          }
        }
      />
      <View>
        <Text
          style={
            Object {
              "fontSize": 13,
              "fontWeight": "bold",
            }
          }
        >
          Freddie Mercuty
        </Text>
        <Text
          style={
            Object {
              "fontSize": 10,
            }
          }
        >
          
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "blue",
        }
      }
    >
      view details
    </Text>
  </View>
</View>
`;

exports[`ListItem should render without an album count 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID="list-item"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 3,
        "marginLeft": 0,
        "marginRight": 0,
        "marginTop": 3,
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 10,
        "shadowColor": "rgba(0, 0, 0, 0.05)",
        "shadowOffset": Object {
          "height": 3,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 3,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "thumb.png",
          }
        }
        style={
          Object {
            "height": 25,
            "marginRight": 10,
            "width": 25,
          }
        }
      />
      <View>
        <Text
          style={
            Object {
              "fontSize": 13,
              "fontWeight": "bold",
            }
          }
        >
          Freddie Mercuty
        </Text>
        <Text
          style={
            Object {
              "fontSize": 10,
            }
          }
        >
          
        </Text>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "blue",
        }
      }
    >
      view details
    </Text>
  </View>
</View>
`;
